src/adapter/src/coord.rs

```diff
@@ -1638,12 +1581,202 @@ impl Coordinator {
         self.apply_local_write(register_ts).await;
     }
 
+    /// Invokes the optimizer on all indexes and materialized views in the catalog and inserts the
+    /// resulting dataflow plans into the catalog state.
+    ///
+    /// `ordered_catalog_entries` must by sorted in dependency order, with dependencies ordered
+    /// before their dependants.
+    ///
+    /// This method does not perform timestamp selection for the dataflows, nor does it create them
+    /// in the compute controller. Both of these steps happen later during bootstrapping.
+    fn bootstrap_dataflow_plans(
+        &mut self,
+        ordered_catalog_entries: &[CatalogEntry],
+    ) -> Result<(), AdapterError> {
+        // The optimizer expects to be able to query its `ComputeInstanceSnapshot` for
+        // collections the current dataflow can depend on. But since we don't yet install anything
+        // on compute instances, the snapshot information is incomplete. We fix that by manually
+        // updating `ComputeInstanceSnapshot` objects to ensure they contain collections previously
+        // optimized.
+        let mut instance_snapshots = BTreeMap::new();
+
+        let optimizer_config = OptimizerConfig::from(self.catalog().system_config());
+
+        for entry in ordered_catalog_entries {
+            let id = entry.id();
+            match entry.item() {
+                CatalogItem::Index(idx) => {
+                    // Collect optimizer parameters.
+                    let compute_instance =
+                        instance_snapshots.entry(idx.cluster_id).or_insert_with(|| {
+                            self.instance_snapshot(idx.cluster_id)
```

Comment:
This is uncovered: https://buildkite.com/materialize/coverage/builds/310#018bb92c-f45a-44a7-a7af-bc5817dd769e
Seems interesting, any chance to cover it in a test?

Reply:
Very surprising that this would be uncovered. This should be alway executed when envd is started with indexes in the catalog (which includes the built-in indexes).

Reply:
Unfortunately not the first time coverage was wrong, thanks for checking. (We have a few open bugs against Rust for that)